### PR TITLE
Add fstring check and error code

### PR DESCRIPTION
**Pull request by yoseft**
Thanks for submitting a PR!

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.
   
Please don't get discouraged as it may take a while to get a review.

### CODE REVIEW

File: src/pydocstyle/violations.py [hunk 1]
```diff
@@ -215,6 +215,7 @@ def to_rst(cls) -> str:
                          'found {0}-quotes')
 D301 = D3xx.create_error('D301', 'Use r""" if any backslashes in a docstring')
 D302 = D3xx.create_error('D302', 'Use u""" for Unicode docstrings')
+D303 = D3xx.create_error('D303', 'F-strings not allowed as docstring')
 
 D4xx = ErrorRegistry.create_group('D4', 'Docstring Content Issues')
```
Comment (Nurdok) [inline src/pydocstyle/violations.py]: docstring**s**

File: src/tests/test_cases/test.py [hunk 2]
```diff
@@ -290,6 +290,20 @@ def first_word_has_unicode_byte():
...
+    def fstring():
+        f"""Do some {eval("'foo'")} and some {eval("'bar'")}."""
+
+    @expect('D303: F-strings not allowed as docstring')
+    def uppercase_fstring():
+        F"""Do some {eval("'foo'")} and some {eval("'bar'")}."""
+
+    @expect('D303: F-strings not allowed as docstring')
+    def raw_fstring():
+        rf"""Do one of: {eval("'foo'")} \ {eval("'bar'")}."""
```
Comment (Nurdok) [inline src/tests/test_cases/test.py]: Also test `fr"docstring"`

File: src/tests/test_cases/test.py [hunk 3]
```diff
@@ -290,6 +290,20 @@ def first_word_has_unicode_byte():
         """あy."""
 
 
+if sys.version_info[0:2] != (3, 5):
```
Comment (Nurdok) [inline src/tests/test_cases/test.py]: This is still failing. I think you need to add an extra file for f-string and only load it if we're with Python 3.6+,

**Comment by samj1912** [comment]
Bump. @yoseft any updates?

**Comment by samj1912** [comment]
Closing this as it has been inactive for a while. Please feel free to re-open if this is still relevant.

**Pull request closed without merging**